pkg/registry/client.go

```diff
@@ -22,7 +23,9 @@ func (t *Client) Fetch(keychain authn.Keychain, repoName string) (v1.Image, stri
 
 	image, err := remote.Image(reference, remote.WithAuthFromKeychain(keychain))
 	if err != nil {
-		return nil, "", err
+		return nil, "", &reconciler.NetworkError{
```

Comment:
Should we attempt to not return a network error on unauthorized?

Reply:
I haven't check the code, but, is not the access verification checked before? is it related to this [PR](https://github.com/pivotal/kpack/pull/1008) ?

Reply:
@matthewmcnew I added some validation to the error throw on `Fetch` and `Save` to verify in the error is not an authorized. Please take a look and let me know your thoughts